### PR TITLE
Pcre2 3194 v3

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3194

Describe changes:
- Use PCRE2 instead of pcre for transform pcrexform
- qa : remove some obsolete files
- pcrexform : checks that we have exactly one substring capture in a signature

This draft PR is about testing PCRE2 support by the CI
After that, the goal is to convert detect-pcre.c to PCRE2

suricata-verify-pr: 480

https://github.com/OISF/suricata-verify/pull/480

### CODE REVIEW

**Comment by catenacyber** [comment]
Fuzzing on coverage finds leaks from `pcre2_compile` even if there is `pcre2_code_free` which should free

**Comment by catenacyber** [comment]
CIFuzz is failing because oss-fuzz uses its own set of dependencies and does not have libpcre2 yet

File: qa/Makefile.am [hunk 1]
```diff
@@ -1,2 +1,2 @@
 SUBDIRS = coccinelle
-EXTRA_DIST = wirefuzz.pl sock_to_gzip_file.py drmemory.suppress
```
Comment (jlucovsky) [inline qa/Makefile.am:2]: Why is this file being deleted?

File: src/detect-transform-pcrexform.c [hunk 2]
```diff
@@ -84,14 +90,36 @@ static int DetectTransformPcrexformSetup (DetectEngineCtx *de_ctx, Signature *s,
...
+                "pcre2 compile of \"%s\" failed at "
+                "offset %d: %s",
+                regexstr, (int)eo, buffer);
         SCFree(pxd);
         SCReturnInt(-1);
     }
+    // check pcd->regex has exactly one capture expression
+    uint32_t nb;
+    if (pcre2_pattern_info(pxd->regex, PCRE2_INFO_CAPTURECOUNT, &nb) < 0) {
+        SCLogError(SC_ERR_INVALID_SIGNATURE, "pcrexform failed getting info about namecount");
```
Comment (jlucovsky) [inline src/detect-transform-pcrexform.c:109]: s/namecount/capturecount

File: qa/Makefile.am (hunk 1 above)
Comment (catenacyber) [reply qa/Makefile.am:2]: 1. Final goal is to replace pcre by pcre2
2. drmemory.suppress references pcre
3. Victor told me this was no longer used and is obsolete

File: src/detect-transform-pcrexform.c (hunk 2 above)
Comment (catenacyber) [reply src/detect-transform-pcrexform.c:109]: Thanks for spotting this

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/5991

**Pull request closed without merging**